src/huggingface_hub/repocard_types.py

```diff
@@ -72,9 +102,9 @@ class SingleResultDataset:
 
 @dataclass
 class SingleResult:
-    metrics: "List[SingleMetric]"
     task: "SingleResultTask"
     dataset: "Optional[SingleResultDataset]"
+    metrics: "List[SingleMetric]"
```

Comment:
For reviewers: this ordering of task > dataset > metrics is more natural and easier to review during Hub PRs